modules/effects/arrow-effects-reactor-data/src/main/kotlin/arrow/effects/reactor/MonoK.kt

```diff
@@ -24,6 +26,13 @@ data class MonoK<A>(val mono: Mono<A>) : MonoKOf<A>, MonoKKindedJ<A> {
   fun <B> map(f: (A) -> B): MonoK<B> =
     mono.map(f).k()
 
+  fun <B> mapFilter(f: (A) -> Option<B>): MonoK<B> =
+    mono
+      .filter { f(it).isDefined() }
+      .map { f(it).orNull()!! }
+      .toMono()
```

Comment:
I don't believe this is a valid instance, as MonoK must always return a value. What do the laws say?

Reply:
I've changed to `flatMap` approach as suggested and all tests are successful!